### PR TITLE
Add optional fields to query Veteran Confirmation

### PR DESCRIPTION
**Pull request by rtravitz**
## Description of change

This PR addresses [#4047](https://github.com/department-of-veterans-affairs/vets-contrib/issues/4047). 

### Notable changes
- Add two optional fields—`middle_name` and `gender`—to query the Veteran Confirmation `/status` endpoint. Having these optional fields present increases the likelihood of a successful MVI query for a given person of interest. 
- Update documentation to include the optional fields

## Testing done

- Added both fields to the happy path payload for the request spec
- Add tests to validate the incoming gender field to comply with the enum that MVI expects

#### Unique to this PR

- [x] API is updated to allow `Gender` and `Middle Name` as optional parameters
- [x] Add and test validations for new optional params
- [x] API docs are updated on the developer portal to reflect this change

#### Applies to all PRs
- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by jeff2d2** [comment]
Looked through the yml and it is exactly how I expected. From a product perspective, this is 💯.

File: modules/veteran_confirmation/VETERAN_CONFIRMATION.yml [hunk 1]
```diff
@@ -29,13 +29,13 @@ info:
...
     #### Required information:
-    * First Name
 
-    * Last Name
 
-    * Date of Birth
 
+    * First Name 
+    * Last Name 
+    * Date of Birth
```
Comment (nfasulo) [inline modules/veteran_confirmation/VETERAN_CONFIRMATION.yml]: Any idea what git is up to here?

File: modules/veteran_confirmation/app/controllers/veteran_confirmation/v0/veteran_status_controller.rb [hunk 2]
```diff
@@ -43,6 +46,13 @@ def valid_ssn?(ssn)
         ssn.is_a?(String) && (all_digits?(ssn) || all_digits_with_hyphens?(ssn))
       end
 
+      def validate_gender
+        gender_options = %w(M F)
+        no_matching_option = params['gender'] && !gender_options.include?(params['gender'])
```
Comment (edmkitty) [inline modules/veteran_confirmation/app/controllers/veteran_confirmation/v0/veteran_status_controller.rb]: Shall we upcase `params['gender']` in the cases of 'm' and 'f'?

File: modules/veteran_confirmation/VETERAN_CONFIRMATION.yml (hunk 1 above)
Comment (jeff2d2) [reply modules/veteran_confirmation/VETERAN_CONFIRMATION.yml]: I see an extra symbol at the end of these lines on master, probably leftovers from copy/pasting from Figma. Thanks for taking care of those @rtravitz!

Comment (rtravitz) [reply modules/veteran_confirmation/VETERAN_CONFIRMATION.yml]: Removed some hidden characters

File: modules/veteran_confirmation/app/controllers/veteran_confirmation/v0/veteran_status_controller.rb (hunk 2 above)
Comment (rtravitz) [reply modules/veteran_confirmation/app/controllers/veteran_confirmation/v0/veteran_status_controller.rb]: Good call. I'll add in the upcasing.

**Pull request merged**